pkg/dockerspec/image.go

```diff
@@ -0,0 +1,50 @@
+package dockerspec
```

Comment:
I wasn't sure what's the best place to put these. As they're more an implementation detail than a part of the API.
The reason I thought creating a separate `pkg` for these is that this could be used by others interacting with these (like buildkit: https://github.com/moby/buildkit/blob/4376f3861b05e0b4a27c2259ee1fa499ac117a41/exporter/containerimage/image/docker_image.go).



Thoughts?

Reply:
Alternatively, we could only have them in the `daemon/containerd` for now.

Reply:
Would these be related to any of these?

- https://github.com/moby/moby/tree/master/image/v1
- https://github.com/moby/moby/tree/master/image/spec

ISTR we had that discussion, where BuildKit now had a definition, but the actual _spec_ is in this repository

Reply:
The discussion in question (I think) https://github.com/rumpl/moby/pull/121#discussion_r1044446599

Reply:
Yes, it's a https://github.com/moby/moby/blob/master/image/spec/v1.2.md (application/vnd.docker.container.image.v1+json).